cmd/moco-init/main.go

```diff
@@ -0,0 +1,178 @@
+		return fmt.Errorf("bad POD_NAME: %s", config.podName)
+	}
+
+	indexUint64, err := strconv.ParseUint(fields[len(fields)-1], 10, 32)
+	if err != nil {
+		return fmt.Errorf("bad POD_NAME %s", config.podName)
+	}
+	config.podIndex = uint32(indexUint64)
+
+	baseUint64, err := strconv.ParseUint(serverIDBase, 10, 32)
+	if err != nil {
+		return fmt.Errorf("invalid server base ID: %s: %w", os.Args[1], err)
+	}
+	config.baseID = uint32(baseUint64)
+
+	_, err = os.Stat(filepath.Join(config.dataDir, initializedFile))
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		if err := initMySQL(mysqld); err != nil {
+			return err
+		}
+	}
+
+	return createConf()
+}
+
+func initMySQL(mysqld string) error {
+	dataDir := filepath.Join(config.dataDir, "data")
+	if err := os.RemoveAll(dataDir); err != nil {
+		return fmt.Errorf("failed to remove dir %s: %w", dataDir, err)
+	}
+
+	var args []string
+	args = append(args, "--basedir="+config.baseDir)
+	args = append(args, "--datadir="+dataDir)
+	args = append(args, "--initialize-insecure")
+
+	if config.lowerCaseTableNames {
+		args = append(args, "--lower_case_table_names=1")
+	}
+
+	cmd := exec.Command(mysqld, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	dotFile := filepath.Join(config.dataDir, "."+initializedFile)
+	if err := os.Remove(dotFile); err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	f, err := os.Create(dotFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if err := unix.Syncfs(int(f.Fd())); err != nil {
+		return fmt.Errorf("failed to sync fs: %w", err)
+	}
+
+	if err := os.Rename(dotFile, filepath.Join(config.dataDir, initializedFile)); err != nil {
+		return err
+	}
+
+	g, err := os.OpenFile(config.dataDir, os.O_RDONLY, 0755)
+	if err != nil {
+		return err
+	}
+	defer g.Close()
+	return g.Sync()
+}
+
+func createConf() error {
+	tmpl := template.Must(template.New("my.cnf").Parse(mycnfTmpl))
+
+	v := struct {
+		ServerID     uint32
+		AdminAddress string
+	}{
+		ServerID:     config.baseID + config.podIndex,
+		AdminAddress: config.podName,
+	}
+
+	f, err := os.OpenFile(filepath.Join(config.confDir, "my.cnf"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to create my.cnf file: %w", err)
+	}
+	defer f.Close()
+
+	if err := tmpl.Execute(f, v); err != nil {
+		return err
+	}
+	return f.Sync()
+}
+
+func init() {
+	rootCmd.Flags().StringVar(&config.baseDir, "base-dir", defaultBaseDir, "The base directory for MySQL.")
+	rootCmd.Flags().StringVar(&config.dataDir, "data-dir", defaultDataDir, "The data directory for MySQL.  Data will be stored in a subdirectory named 'data'")
+	rootCmd.Flags().StringVar(&config.confDir, "conf-dir", defaultConfDir, "The directory where configuration file is created.")
+	rootCmd.Flags().BoolVar(&config.lowerCaseTableNames, "lower-case-table-names", false, "Initialize mysqld with 'lower-case-table-names=1'.")
```

Comment:
Could you input the value of `lower-case-table-names` flags as a string?
In the specification, MySQL accepts `0`, `1`, or `2` as `lower-case-table-names`.
https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html

So I think it's better that the moco-init accepts same values.

Reply:
Could you check the behavior of `mysql --initialize-insecure`, when the invalid value of `lower-case-table-names` is specified?
If the MySQL gets an error, we need not validate the value of `lower-case-table-names`.

Reply:
```console
$ docker run -it quay.io/cybozu/mysql:8.0.30 --lower_case_table_names=11
2022-10-05T02:15:46.035781Z 0 [Warning] [MY-000081] [Server] option 'lower_case_table_names': unsigned value 11 adjusted to 2.
```

It doesn't seem to be an error.
I will add a validation.

Reply:
added: https://github.com/cybozu-go/moco-agent/pull/71/commits/602e10f88bbbd5305afa046c949099422a723b48